tools/pythonpkg/src/pyexpression.cpp

```diff
@@ -0,0 +1,345 @@
+}
+
+shared_ptr<DuckDBPyExpression> DuckDBPyExpression::GreaterThanOrEqual(const DuckDBPyExpression &other) {
+	return ComparisonExpression(ExpressionType::COMPARE_GREATERTHANOREQUALTO, *this, other);
+}
+
+shared_ptr<DuckDBPyExpression> DuckDBPyExpression::LessThan(const DuckDBPyExpression &other) {
+	return ComparisonExpression(ExpressionType::COMPARE_LESSTHAN, *this, other);
+}
+
+shared_ptr<DuckDBPyExpression> DuckDBPyExpression::LessThanOrEqual(const DuckDBPyExpression &other) {
+	return ComparisonExpression(ExpressionType::COMPARE_LESSTHANOREQUALTO, *this, other);
+}
+
+// AND, OR and NOT
+
+shared_ptr<DuckDBPyExpression> DuckDBPyExpression::Not() {
+	return DuckDBPyExpression::InternalUnaryOperator(ExpressionType::OPERATOR_NOT, *this);
+}
+
+shared_ptr<DuckDBPyExpression> DuckDBPyExpression::And(const DuckDBPyExpression &other) {
+	return DuckDBPyExpression::InternalConjunction(ExpressionType::CONJUNCTION_AND, *this, other);
+}
+
+shared_ptr<DuckDBPyExpression> DuckDBPyExpression::Or(const DuckDBPyExpression &other) {
+	return DuckDBPyExpression::InternalConjunction(ExpressionType::CONJUNCTION_OR, *this, other);
+}
+
+// IN
+
+shared_ptr<DuckDBPyExpression> DuckDBPyExpression::In(const py::args &args) {
+	vector<unique_ptr<ParsedExpression>> expressions;
+	expressions.reserve(args.size() + 1);
+	expressions.push_back(GetExpression().Copy());
+
+	for (auto arg : args) {
+		shared_ptr<DuckDBPyExpression> py_expr;
+		if (!py::try_cast<shared_ptr<DuckDBPyExpression>>(arg, py_expr)) {
+			throw InvalidInputException("Please provide arguments of type Expression!");
+		}
+		auto expr = py_expr->GetExpression().Copy();
+		expressions.push_back(std::move(expr));
+	}
+	auto operator_expr = make_uniq<OperatorExpression>(ExpressionType::COMPARE_IN, std::move(expressions));
+	return make_shared<DuckDBPyExpression>(std::move(operator_expr));
+}
+
+shared_ptr<DuckDBPyExpression> DuckDBPyExpression::NotIn(const py::args &args) {
+	auto in_expr = In(args);
+	return in_expr->Not();
+}
+
+// Order modifiers
+
+shared_ptr<DuckDBPyExpression> DuckDBPyExpression::Ascending() {
+	auto py_expr = Copy();
+	py_expr->order_type = OrderType::ASCENDING;
+	return py_expr;
+}
+
+shared_ptr<DuckDBPyExpression> DuckDBPyExpression::Descending() {
+	auto py_expr = Copy();
+	py_expr->order_type = OrderType::DESCENDING;
+	return py_expr;
+}
+
+// Null order modifiers
+
+shared_ptr<DuckDBPyExpression> DuckDBPyExpression::NullsFirst() {
+	auto py_expr = Copy();
+	py_expr->null_order = OrderByNullType::NULLS_FIRST;
+	return py_expr;
+}
+
+shared_ptr<DuckDBPyExpression> DuckDBPyExpression::NullsLast() {
+	auto py_expr = Copy();
+	py_expr->null_order = OrderByNullType::NULLS_LAST;
+	return py_expr;
+}
+
+// Unary operators
+
+shared_ptr<DuckDBPyExpression> DuckDBPyExpression::Negate() {
+	vector<unique_ptr<ParsedExpression>> children;
+	children.push_back(GetExpression().Copy());
+	return DuckDBPyExpression::InternalFunctionExpression("-", std::move(children), true);
```

Comment:
can we have an assert that there is only one child here? Also, why the `"-"` character for negate? Isn't `"!"` more common?

Reply:
`!` would create a boolean, this just inverses the number I think?